messaging/kafka-producer/src/test/java/io/quarkus/ts/messaging/kafka/producer/OpenShiftKafkaSnappyTest.java

```diff
@@ -0,0 +1,67 @@
+package io.quarkus.ts.messaging.kafka.producer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.Duration;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.KafkaService;
+import io.quarkus.test.bootstrap.Protocol;
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.annotations.DisabledOnNative;
+import io.quarkus.test.services.KafkaContainer;
+import io.quarkus.test.services.QuarkusApplication;
+import io.quarkus.test.services.containers.model.KafkaVendor;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.helpers.test.UniAssertSubscriber;
+import io.vertx.mutiny.core.buffer.Buffer;
+
+@DisabledOnNative(reason = "In native mode, Snappy is disabled by default as the use of Snappy requires embedding a native library and unpacking it when the application starts.")
+public class OpenShiftKafkaSnappyTest {
+
+    private static final int TIMEOUT_SEC = 5;
+    private static final String FILTER_COMMAND_LOG_CONTAINER = "/bin/bash -c \"./bin/kafka-run-class.sh kafka.tools.DumpLogSegments --deep-iteration --print-data-log --files /tmp/kafka-logs/test-0/00000000000000000000.log | head\"";
+
+    @KafkaContainer(vendor = KafkaVendor.STRIMZI)
+    static final KafkaService kafka = new KafkaService().withProperty("auto.create.topics.enable", "false");
+
+    @QuarkusApplication
+    static RestService app = new RestService().withProperty("kafka.bootstrap.servers", kafka::getBootstrapUrl);
+
+    @Test
+    public void checkCompressCodecSnappy() throws IOException {
+        String msg = "This is the message";
+
+        UniAssertSubscriber<Object> subscriber = makeHttpReqWithMessage("/messageEvent", msg)
+                .subscribe().withSubscriber(UniAssertSubscriber.create());
+        subscriber.awaitItem(Duration.ofSeconds(TIMEOUT_SEC)).getItem();
+
+        ProcessBuilder pbGetPods = new ProcessBuilder("oc", "get", "pods", "-l", "app=kafka", "-o",
+                "jsonpath={.items[0].metadata.name}");
+        Process processGetPods = pbGetPods.start();
+        String kafkaPodName = new BufferedReader(new InputStreamReader(processGetPods.getInputStream())).readLine().trim();
+
+        ProcessBuilder pbExecCommand = new ProcessBuilder("oc", "rsh", kafkaPodName, "bash", "-c",
+                FILTER_COMMAND_LOG_CONTAINER);
+        Process processExecCommand = pbExecCommand.start();
+        String logSegmentHead = new BufferedReader(new InputStreamReader(processExecCommand.getInputStream())).lines()
```

Comment:
IMHO it would be easier to to wait for process to end, using pbExecCommand.waitFor() and then read the result using something like `new String(process.getInputStream().readAllBytes());`

Reply:
Yeah, I think `waitFor` is a good point here.